mycroft/client/wifisetup/main.py

```diff
@@ -61,17 +88,62 @@ def sysctrl(*args):
     return cli('systemctl', *args)
 
 
-class WebServer(Thread):
-    DIR = dirname(realpath(__file__))
+class MycroftHTTPRequestHandler(SimpleHTTPRequestHandler):
+
+    def do_HEAD(self):
+        LOG.info("do_HEAD being called....")
+        if not self.redirect():
+            SimpleHTTPRequestHandler.do_HEAD(self)
+
+    def do_GET(self):
+        LOG.info("do_GET being called....")
+        if not self.redirect():
+            SimpleHTTPRequestHandler.do_GET(self)
+
+    def redirect(self):
+        try:
+            LOG.info("***********************")
+            LOG.info("**   HTTP Request   ***")
+            LOG.info("***********************")
+            LOG.info("Requesting: "+self.path)
+            LOG.info("REMOTE_ADDR:"+self.client_address[0])
+            LOG.info("SERVER_NAME:"+self.server.server_address[0])
+            LOG.info("SERVER_PORT:"+str(self.server.server_address[1]))
+            LOG.info("SERVER_PROTOCOL:"+self.request_version)
+            LOG.info("HEADERS...")
+            LOG.info(self.headers)
+            LOG.info("***********************")
+
+            # path = self.translate_path(self.path)
+            if "mycroft.ai" in self.headers['host']:
+                LOG.info("No redirect")
+                return False
+            else:
+                LOG.info("303 redirect to http://start.mycroft.ai")
+                self.send_response(303)
+                self.send_header("Location", "http://start.mycroft.ai")
+                self.end_headers()
+                return True
+        except:
+            tb = traceback.format_exc()
+            LOG.info("exception caught")
+            LOG.info(tb)
+            return False
 
+
+class WebServer(Thread):
     def __init__(self, host, port):
         super(WebServer, self).__init__()
         self.daemon = True
+        LOG.info("Creating TCPServer...")
+        # self.server = TCPServer((host, port), MycroftHTTPRequestHandler)
```

Comment:
There was some stuff left behind while this was still in development.  It wasn't being kept around for legacy reasons, it was potentially going to be brought into play.  (In fact, MycroftHTTPRequestHandler is being used in my latest commits to enable automatic display of the portal on an iOS device.)